charts/seed-bootstrap/charts/monitoring/templates/config.yaml

```diff
@@ -41,13 +41,18 @@ data:
         insecure_skip_verify: true
       kubernetes_sd_configs:
       - role: pod
+        namespaces:
+            names: [ garden ]
       relabel_configs:
       - source_labels:
-        - __meta_kubernetes_namespace
         - __meta_kubernetes_pod_annotation_prometheus_io_scrape
         - __meta_kubernetes_pod_annotation_prometheus_io_port
-        regex: garden;true;(.+)
+        regex: true;(.+)
         action: keep
+      - source_labels:
+        - __meta_kubernetes_pod_label_role
+        regex: gardenlet
+        action: drop
```

Comment:
Instead of excluding the gardenlet explicitly here, you could remove the `prometheus.io/*` annotations from its pods.
Right now this is inconsistent with the other `garden` namespace components (grm, vpa, etc.)

Reply:
Addressed in https://github.com/gardener/gardener/pull/7180/commits/d6d468a3e02f9d4cd78cb357cb74815342940003